### ISSUE TITLE
Convert test suite to py.test

### ISSUE THREAD

**Issue by drkjam** [drkjam/netaddr#98]
This is a ticket to track work currently taking on the `netaddr_pytest` branch.

**Comment by drkjam** [comment]
@snordhausen could you review the pull request for the branch I've been working on and let me know if it's OK to merge?

**Comment by drkjam** [comment]
Overview of changes
----------------------------
Ported the old style doctest suite to individual `py.test` unit tests and moved tests up one level in the source tree (they will no longer be included in the release packages).

Kept doctest tutorials as is but only for Python 2.x. you can still run these as tests (to ensure they are working using the `run_doctests.py` script in the `tutorials` directory.

Once this ticket is completed I'll go ahead and release `0.7.15` (barring any urgent outstanding issues).

**Comment by snordhausen** [comment]
@drkjam It seems the tests are not compatible with people that use virtual environments:
```
platform linux2 -- Python 2.7.6 -- pytest-2.4.2
collected 504 items / 6 errors 

test/core/test_compat.py s......
test/core/test_pubsub.py .
test/eui/test_eui.py ...s................
test/eui/test_ieee_parsers.py ..ss
test/ip/test_cidr_v4.py .........
test/ip/test_cidr_v6.py ....
test/ip/test_dns.py ..
test/ip/test_ip_categories.py ........
test/ip/test_ip_comparisons.py ..
test/ip/test_ip_globs.py .......
test/ip/test_ip_ranges.py ................
test/ip/test_ip_rfc1924.py .
test/ip/test_ip_sets.py ............................
test/ip/test_ip_v4.py ...............................ss.....
test/ip/test_ip_v4_v6_conversions.py .....
test/ip/test_ip_v6.py ............
test/ip/test_network_ops.py ...
test/ip/test_nmap.py ....
test/ip/test_old_specs.py ...
test/ip/test_platform_osx.py s.
test/ip/test_socket_module_fallback.py .................
test/strategy/test_eui48_strategy.py ..s.
test/strategy/test_ipv4_strategy.py ..s..
test/strategy/test_ipv6_strategy.py ..s...............................................
venv/lib/python2.7/site-packages/setuptools/tests/test_bdist_egg.py ..
venv/lib/python2.7/site-packages/setuptools/tests/test_build_ext.py .
venv/lib/python2.7/site-packages/setuptools/tests/test_develop.py .
venv/lib/python2.7/site-packages/setuptools/tests/test_dist_info.py ..
venv/lib/python2.7/site-packages/setuptools/tests/test_easy_install.py ............
venv/lib/python2.7/site-packages/setuptools/tests/test_markerlib.py .
venv/lib/python2.7/site-packages/setuptools/tests/test_packageindex.py ......F........
venv/lib/python2.7/site-packages/setuptools/tests/test_resources.py ..............................
venv/lib/python2.7/site-packages/setuptools/tests/test_sandbox.py ..
venv/lib/python2.7/site-packages/setuptools/tests/test_sdist.py .....FF.
venv/lib/python2.7/site-packages/setuptools/tests/test_svn.py ..FFFFFFFFFFFFFFFFFFF.
venv/lib/python2.7/site-packages/setuptools/tests/test_test.py .
venv/lib/python2.7/site-packages/setuptools/tests/test_upload_docs.py .
venv/lib/python2.7/site-packages/wheel/test/test_basic.py ......F..
venv/lib/python2.7/site-packages/wheel/test/test_install.py ..
venv/lib/python2.7/site-packages/wheel/test/test_keys.py ..
venv/lib/python2.7/site-packages/wheel/test/test_paths.py .
venv/lib/python2.7/site-packages/wheel/test/test_ranking.py .
venv/lib/python2.7/site-packages/wheel/test/test_signatures.py ...
venv/lib/python2.7/site-packages/wheel/test/test_tagopt.py .......
venv/lib/python2.7/site-packages/wheel/test/test_tool.py F
venv/lib/python2.7/site-packages/wheel/test/test_wheelfile.py ..
venv/local/lib/python2.7/site-packages/setuptools/tests/test_bdist_egg.py ..
venv/local/lib/python2.7/site-packages/setuptools/tests/test_build_ext.py .
venv/local/lib/python2.7/site-packages/setuptools/tests/test_develop.py .
venv/local/lib/python2.7/site-packages/setuptools/tests/test_dist_info.py ..
venv/local/lib/python2.7/site-packages/setuptools/tests/test_easy_install.py ............
venv/local/lib/python2.7/site-packages/setuptools/tests/test_markerlib.py .
venv/local/lib/python2.7/site-packages/setuptools/tests/test_packageindex.py ......F........
venv/local/lib/python2.7/site-packages/setuptools/tests/test_resources.py .............................F
venv/local/lib/python2.7/site-packages/setuptools/tests/test_sandbox.py ..
venv/local/lib/python2.7/site-packages/setuptools/tests/test_sdist.py .....FF.
venv/local/lib/python2.7/site-packages/setuptools/tests/test_svn.py ..FFFFFFFFFFFFFFFFFFF.
venv/local/lib/python2.7/site-packages/setuptools/tests/test_test.py .
venv/local/lib/python2.7/site-packages/setuptools/tests/test_upload_docs.py .
venv/local/lib/python2.7/site-packages/wheel/test/test_basic.py
```
It seems that the test framework is picking up somebody else's test cases that happen to be in the virtual environment. However, I have to use a virtual environment because otherwise stuff like "pip install sphinx" will not work. Or I'd have to develop as root all the time.

To reproduce, do a "virtualenv venv" in the netaddr directory. You don't even have to enter the venv, its mere existence causes the tests to fail.

**Comment by snordhausen** [comment]
I also discovered a Python 3.5 issue:

Old test framework
* Python 3.4.0 works
* Python 3.5 beta 2 works

New test framework
* Python 3.4.0 works
* Python 3.5 beta 2 does NOT work

But that seems to be a pytest vs. Python 3.5 issue, not related to your changes. It seems some AST interface was changed, the backtrace ends with
```
/usr/local/lib/python3.5/ast.py:245: in visit
    return visitor(node)
_pytest.assertion.rewrite:521: in visit_Name
    ???
```

Apart from that issue, I really do like the
> 6,234 additions and 9,404 deletions.

part of the pull request :+1:

**Comment by snordhausen** [comment]
The Python 3.5 issue seems to be a rediscovery of https://github.com/pytest-dev/pytest/issues/744

**Comment by drkjam** [comment]
Thanks for the prompt feedback. Only netaddr tests run now if you have a local virtualenv in the root dir. I'm happy to wait for pytest dev team to fix this issue. Branch merged into 0.7.x mainline.
